### PR TITLE
Add support for slack_channel_id to `resource_service`

### PR DESCRIPTION
**Pull request by mohammadmuazam**
Ref: #50

### CODE REVIEW

File: internal/api/services.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ type Service struct {
 	Dependencies       []string           `json:"depends" tf:"dependencies"`
 	ActiveAlertSources map[string]string  `json:"-" tf:"active_alert_source_endpoints"`
 	AlertSources       map[string]string  `json:"-" tf:"alert_source_endpoints"`
+	Slack              *SlackChannel      `json:"slack" tf:"-"`
+}
+
+type SlackChannel struct {
+	ChannelID string `json:"channel_id" tf:"-"`
```
Comment (roshan8) [inline internal/api/services.go:32]: why tf:"-"`?
doesn't it ignore saving state?

File: internal/api/services.go [hunk 2]
```diff
@@ -127,3 +140,8 @@ func (client *Client) DeleteService(ctx context.Context, id string) (*any, error
 	url := fmt.Sprintf("%s/services/%s", client.BaseURLV3, id)
 	return Request[any, any](http.MethodDelete, url, client, ctx, nil)
 }
+
+func (client *Client) UpdateSlackChannel(ctx context.Context, serviceID string, req *AddSlackChannelReq) (*any, error) {
```
Comment (roshan8) [inline internal/api/services.go:144]: Does creating service support slack channel setup?
Making 2 api calls per resource is considered an anti-pattern in tf.

**Review by roshan8 — CHANGES_REQUESTED** [review]

File: internal/api/services.go (hunk 1 above)
Comment (mohammadmuazam) [reply internal/api/services.go:32]: Check L62-64, we are storing it there

File: internal/api/services.go (hunk 2 above)
Comment (mohammadmuazam) [reply internal/api/services.go:144]: No, we don't support that

Comment (roshan8) [reply internal/api/services.go:144]: ok

**Review by roshan8 — APPROVED** [review]

**Pull request merged**